clang/include/clang/Basic/Attr.td

```diff
@@ -1289,11 +1289,28 @@ def LoopUnrollHint : InheritableAttr {
 
 def IntelReqdSubGroupSize: InheritableAttr {
   let Spellings = [GNU<"intel_reqd_sub_group_size">, CXX11<"cl", "intel_reqd_sub_group_size">];
-  let Args = [ExprArgument<"SubGroupSize">];
+  let Args = [ExprArgument<"SubGroupSizeExpr">];
   let Subjects = SubjectList<[Function, CXXMethod], ErrorDiag>;
   let Documentation = [IntelReqdSubGroupSizeDocs];
   let LangOpts = [OpenCL, SYCLIsDevice, SYCLIsHost];
   let PragmaAttributeSupport = 0;
+  let AdditionalMembers = [{
+  private:
+    int SubGroupSize = 0;
+
+  public:
+    int getSubGroupSizeEvaluated(ASTContext &Ctx) {
```

Comment:
That is an interesting approach. I was just about refactoring FPGA specific attributes and may adopt it.

Do you consider if it's possible/worth to move this function (with renaming) to InheritableAttr or even create a new class, like SYCLFunctionAttr?

Reply:
I mean, we have tons of attributes with similar routines.

Reply:
> Do you consider if it's possible/worth to move this function (with renaming) to InheritableAttr or even create a new class, like SYCLFunctionAttr?

Good point, I will take a look at FPGA-related attributes and propose something more common/generic in this PR

Reply:
It's not necessary to do it right now - currently FPGA attributes doesn't support template instantiation of arguments (add support for it is the main purpose of the patch I'm preparing).

Reply:
@MrSidims, see 608ee3a0b636bec4ff95a2afb4f29d8aa4c0253e - I've introduced new type of attribute argument, which should be re-usable by others as well.

> add support for it is the main purpose of the patch I'm preparing

Could you please try to embed my PR into your patch to see if proposed way can actually be re-used by someone else and works fine?

Reply:
Just tried it out. It works perfectly well with num_simd_work_items attribute.